File: examples/custom_http_routes.cs
// Copy into your C# host Program.cs (requires FusionFramework reference).
//
// Routes:
//   GET /api/user              -> Get()
//   GET /api/user/test/user    -> UserAction()   ([action] -> user)

using FusionFramework;

[Route("/api/[module]", Tags = new[] { "users" })]
public class UserModule : FusionBaseApi
{
    public object Get() =>
        Response(new { mode = "convention" }, Status.HTTP_SUCCESS);

    [HttpGet("test/[action]", Title = "User action")]
    public object UserAction() =>
        Response(new { mode = "custom", action = "user" }, Status.HTTP_SUCCESS);
}

// Program entry:
// Route.Register<UserModule>();
// SettingsStore.Current.EnsureLoaded(Directory.GetCurrentDirectory());
// using var app = new FusionApp(SettingsStore.GetSettings());
// app.Listen();

File: examples/custom_http_routes.py
"""Custom HTTP method routes — convention get/post plus @http_get handlers."""

from fusion_framework import status
from fusion_framework.api import FusionBaseApi
from fusion_framework.app import FusionApp
from fusion_framework.config import get_settings, load_settings_module
from fusion_framework.http_route import http_get
from fusion_framework.route import route


@route("/api/[module]", tags=["users"])
class UserModule(FusionBaseApi):
    # Convention route: GET /api/user
    def get(self):
        return self.response({"mode": "convention"}, status=status.HTTP_SUCCESS)

    # Custom route: GET /api/user/test/user  ([action] -> user)
    @http_get("test/[action]", title="User action")
    def UserAction(self):
        return self.response({"mode": "custom", "action": "user"}, status=status.HTTP_SUCCESS)


def main() -> None:
    load_settings_module("settings")
    FusionApp(get_settings()).listen()


if __name__ == "__main__":
    main()
